pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
+                #     choose one place to set")
+                # else:
+                self._precision_flag = self._strategy_flag._precision_plugin
+            if self._strategy_flag._checkpoint_io:
+                if self.checkpoint_io:
                     raise MisconfigurationException(
-                        "You can only specify one precision plugin."
-                        f" Available: {type(precision).__name__}, given: {type(plug).__name__}"
+                        "checkpoint_io set through both strategy class and plugins, choose one"
                     )
-            elif isinstance(plug, CheckpointIO):
-                if checkpoint is None:
-                    checkpoint = plug
                 else:
+                    self.checkpoint_io = self._strategy_flag._checkpoint_io
+            if getattr(self._strategy_flag, "cluster_environment", None):
+                if self._cluster_environment:
                     raise MisconfigurationException(
-                        "You can only specify one checkpoint plugin."
-                        f" Available: {type(checkpoint).__name__}, given: {type(plug).__name__}"
+                        "cluster_environment set through both strategy class and plugins, choose one"
                     )
-            elif isinstance(plug, ClusterEnvironment):
-                if cluster_environment is None:
-                    cluster_environment = plug
                 else:
-                    raise MisconfigurationException(
-                        "You can only specify one cluster environment. Found more than 1 cluster environment plugin"
-                    )
-            else:
+                    self._cluster_environment = getattr(self._strategy_flag, "cluster_environment")
+            # RFC existing accel_conn doesn't handle this, should we add conflict check?
+            # eg: parallel_device is torch.device(cpu) but accelerator=gpu
+            if hasattr(self._strategy_flag, "parallel_devices"):
+                if self._strategy_flag.parallel_devices:
+                    if self._strategy_flag.parallel_devices[0].type == "cpu":
+                        self._accelerator_flag = "cpu"
+                    if self._strategy_flag.parallel_devices[0].type == "cuda":
+                        self._accelerator_flag = "gpu"
+
+        amp_type = amp_type.lower() if isinstance(amp_type, str) else None
+        self._amp_type_flag = AMPType.from_str(amp_type)
+
+        if amp_level is not None and self._amp_type_flag != AMPType.APEX:
+            raise MisconfigurationException(
+                f"You have asked for `amp_level={amp_level!r}` but it's only supported with `amp_backend='apex'`."
+            )
+        self._amp_level_flag = amp_level
+
+    def _device_config_check_and_set_final_flags(self, devices, num_nodes, num_processes, gpus, ipus, tpu_cores):
+        if num_nodes == "auto":
```

Comment:
where is thsi check coming from? i don't see any reference to num_nodes being auto

Reply:
Added this to address test failure. I will double check

Reply:
Nope, will remove and test whether the tests will pass